src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -438,7 +429,6 @@ void WorldSession::HandleCharCreateOpcode(WorldPacket& recvData)
             }
             bool haveSameRace = false;
             uint32 heroicReqLevel = sWorld->getIntConfig(CONFIG_CHARACTER_CREATING_MIN_LEVEL_FOR_HEROIC_CHARACTER);
-            bool hasHeroicReqLevel = (heroicReqLevel == 0);
```

Comment:
I seem to have found the error
We should remove it first
UpdateAccountFlag(ACCOUNT_FLAG_DEATH_KNIGHT_OK, true);